Update Mac install instructions (0.51+)

avanwinkle opened the pull request:
This PR renovates the installation instructions for Mac, cleaning up some legacy workarounds and eliminating some steps. The steps were written based on a fresh wipe/install of High Sierra. I made (at least) two test runs each of latest release (`pip install mpf`), dev release (`--pre`), and dev local (`-e`) on Python 3.6 and 3.7. 

A few questions and notes about the changes:

* The new instructions are based on using virtual environments, which are included in Python 3. I believe as a result of the virtual environments some of the sudo elevations are no longer necessary

* Pip is included in Python 3 and its virtual environments, so the manual pip installation is removed

* As of 0.51, `setuptools` and `cython` seem to play nice without explicit versioning.

* What does `PyQt5` do? I didn't install it on any of my test runs and all is working well.

* I didn't find the Terminal patch necessary, as the UI rendered properly in both native Terminal and iTerm.

* I found the best success doing the MPF/MC install and _then_ updating Kivy from 1.10 to 1.11. Trying to do Kivy first led to grossness.


When I was playing around with installation steps and particularly Kivy, I would sometimes get the SDL `One of the two will be used. Which one is undefined.` warnings. But when I do fresh virtual environments following the final version of the instructions, I don't. So... that's good?

avanwinkle commented:
Travis build is failing on matrix lights documentation, looks like it doesn't like the colon:
```
ValueError: invalid literal for int() with base 10: '2:10'
```

I tried scoping through the other docs but couldn't find anything that suggested a different format for the old P-ROC matrix lights. Thoughts?

jabdoa2 commented on install/mac.rst:
```diff
@@ -162,39 +165,123 @@ different between the two:
 .. code-block:: console
 
-    pip3 install setuptools cython==0.25.2 --upgrade
+    pip install --upgrade setuptools cython
```
I guess we need "pip3 install -I Cython==0.27.3". See https://github.com/missionpinball/mpf-debian-installer/blob/dev/install-mc-dependencies#L11

avanwinkle replied:
Are there concerns about future versions? I've got 0.29.3 in both my 3.6 and 3.7 versions and they work just fine.

jabdoa2 replied:
It really only depends on kivy. They document which versions will work

jabdoa2 commented on install/mac.rst:
```diff
@@ -308,13 +374,54 @@ this:
+6. Update Kivy to 1.11
+----------------------
+
+There is a known bug in Kivy 1.10 that causes intermittent freezing in applications, including MC.
```
They actually released 1.10.1 which contains the fix. That should be our version.

avanwinkle replied:
Cool, I'll remove this whole section then. One less step!

jabdoa2 commented on install/mac.rst:
```diff
@@ -308,13 +374,54 @@ this:
+5.2 Install MPF & MC (Latest)
+-----------------------------
+
+The stable release of MPF is updated every few months, but if you always want the most up-to-date
```
Not sure if dev is right for most users. Maybe we could add a warning?

avanwinkle replied:
Done!

jabdoa2 commented:
I will have a look at the tests. Probably related to the update of ruamel which I did.

The SDL error worries me. Peter is looking into using the DMG to install MPF together with the kivy app to get rid of that.

avanwinkle commented:
The SDL error is a bit confusing, because it only appears when kivy is installed from a wheel (generally as an mpf dependency). Installing kivy directly from a zip archive does not cause the SDL error.

```
pip install https://github.com/kivy/kivy/archive/stable-1.10.zip
```

avanwinkle commented:
Updated with less Kivy complication; just installing it with `--no-binary` eliminates the SDL errors. 

The MPF Monitor installation instructions include PyQt5, so I've adjusted those for Mac and will keep PyQt5 out of the main instructions.
